patches/tModLoader/Terraria/Projectile.cs.patch

```diff
@@ -414,16 +414,66 @@
  			if (wet)
  				position += wetVelocity;
  			else
-@@ -13080,6 +_,9 @@
+@@ -13006,6 +_,7 @@
+ 
+ 			fisher.playerFishingConditions = Main.player[owner].GetFishingConditions();
+ 			int baitItemType = fisher.playerFishingConditions.BaitItemType;
++
+ 			if (baitItemType == 2673) {
+ 				Main.player[owner].displayedFishingInfo = Language.GetTextValue("GameUI.FishingWarning");
+ 				if ((fisher.X < 380 || fisher.X > Main.maxTilesX - 380) && fisher.waterTilesCount > 1000 && !NPC.AnyNPCs(370)) {
+@@ -13080,12 +_,23 @@
  			FishingCheck_RollEnemySpawns(ref fisher);
  			FishingCheck_RollItemDrop(ref fisher);
  			bool flag = false;
-+			PlayerHooks.CatchFish(Main.player[owner], Main.player[owner].inventory[Main.player[owner].selectedItem],
-+				fisher.waterNeededToFish, fisher.inLava ? 1 : fisher.inHoney ? 2 : 0, fisher.fishingLevel, fisher.heightLevel, fisher.questFish, ref fisher.rolledItemDrop);
++			string sonarText = null;
++			Color color = Color.White;
++			PlayerHooks.CatchFish(Main.player[owner], fisher, ref sonarText, ref color);
 +
  			if (fisher.rolledItemDrop > 0) {
- 				if (Main.player[owner].sonarPotion) {
- 					Item item = new Item();
++				if (Main.player[owner].sonarPotion && (sonarText == null || sonarText.Trim().Length > 0)) {
++
+-				if (Main.player[owner].sonarPotion) {
++					if (sonarText == null) {
+-					Item item = new Item();
++						Item item = new Item();
+-					item.SetDefaults(fisher.rolledItemDrop);
++						item.SetDefaults(fisher.rolledItemDrop);
+-					item.position = position;
++						item.position = position;
+-					PopupText.sonarText = PopupText.NewText(PopupTextContext.SonarAlert, item, 1, noStack: true);
++						PopupText.sonarText = PopupText.NewText(PopupTextContext.SonarAlert, item, 1, noStack: true);
++					}
++					else {
++						PopupText.sonarText = PopupText.NewText(PopupTextContext.SonarAlert, sonarText, color, base.Center, stay5TimesLonger: false);
++					}
++
+ 					if (PopupText.sonarText > -1)
+ 						Main.popupText[PopupText.sonarText].sonar = true;
+ 				}
+@@ -13093,13 +_,20 @@
+ 				float num3 = fisher.fishingLevel;
+ 				ai[1] = (float)Main.rand.Next(-240, -90) - num3;
+ 				localAI[1] = fisher.rolledItemDrop;
++				localAI[0] = fisher.rolledItemStack;
+ 				netUpdate = true;
```

Comment:
Is this change/extension to the AI necessary? It's not mentioned in the PR but is definitely worth mentioning if this goes through (on the wiki later)

Reply:
I needed some space to allow for item stack manipulation to be possible with this system. Normal Vanilla behaviour is to always use stack of 1, and fix that in the player catch method.

localAI[0] was only being written to, but never read, so I used that to transmit the stack. Not my ideal solution, but there was no more space to use.

Reply:
Acceptable, but please put a comment in the code